### PR TITLE
Tweak: Update Onboarding module

### PR DESCRIPTION
**Pull request by udidol**

### CODE REVIEW

File: core/app/modules/onboarding/module.php [hunk 1]
```diff
@@ -192,30 +193,46 @@ private function maybe_update_site_name() {
...
-			];
+		$attachment_url = wp_get_attachment_url( $data['attachmentId'] );
+
+		// Check if the attachment exists. If it does not, exit here.
+		if ( ! $attachment_url ) {
+			return $data_error;
 		}
 
-		return $result;
+		set_theme_mod( 'custom_logo', $data['attachmentId'] );
```
Comment (KingYes) [inline core/app/modules/onboarding/module.php:228]: Use value from `absint()`

File: core/app/modules/onboarding/module.php [hunk 2]
```diff
@@ -305,19 +324,21 @@ private function activate_hello_theme() {
 	 * @return array
 	 */
 	private function upload_and_install_pro() {
+		if ( ! current_user_can( 'install_plugins' ) ||  ! current_user_can( 'activate_plugins' ) ) {
```
Comment (KingYes) [inline core/app/modules/onboarding/module.php:327]:
```suggestion
		if ( ! current_user_can( 'install_plugins' ) || ! current_user_can( 'activate_plugins' ) ) {
```

**Review by KingYes — CHANGES_REQUESTED** [review]

File: core/app/modules/onboarding/module.php [hunk 3]
```diff
@@ -192,30 +193,46 @@ private function maybe_update_site_name() {
 	 * @return array
 	 */
 	private function maybe_update_site_logo() {
+		// Validate capability
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/module.php:196]: Done.

File: core/app/modules/onboarding/module.php [hunk 4]
```diff
@@ -192,30 +193,46 @@ private function maybe_update_site_name() {
...
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return $this->permission_error_response;
+		}
+
+		$data_error = [
+			'status' => 'error',
+			'payload' => [
+				'error_message' => esc_html__( 'there was a problem setting your site logo', 'elementor' ),
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/module.php:204]: Done.

File: core/app/modules/onboarding/module.php [hunk 5]
```diff
@@ -418,6 +439,13 @@ private function maybe_handle_ajax() {
 	}
 
 	public function __construct() {
+		$this->permission_error_response = [
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/app/modules/onboarding/module.php:442]: Done. Used a method that returns the array.

File: core/app/modules/onboarding/module.php (hunk 1 above)
Comment (udidol) [reply core/app/modules/onboarding/module.php:228]: Done.

File: core/app/modules/onboarding/module.php (hunk 2 above)
Comment (udidol) [reply core/app/modules/onboarding/module.php:327]: Done.

**Review by KingYes — APPROVED** [review]

**Pull request merged**